### PR TITLE
Minor style fixes / refactoring

### PR DESCRIPTION
**Pull request by Geod24**
See each commit.

### CODE REVIEW

File: source/geod24/LocalRest.d [hunk 1]
```diff
@@ -303,10 +303,12 @@ private class Connection
...
-                while (this.resp_chn.read(res, timeout))
-                    if (res.id == resp_id)
-                        return;
-                res = Response(Status.Timeout, resp_id, SerializedData("Request timed-out"));
+                if (!this.resp_chn.read(res, timeout))
+                    res = Response(Status.Timeout, resp_id, SerializedData("Request timed-out"));
+                // Otherwise the `res` variable would have been filled by `read`,
+                // and the response ID should match since the main thread never
+                // send concurrent requests
+                assert(res.id == resp_id);
```
Comment (omerfirmak) [inline source/geod24/LocalRest.d]: Could we hit a case similar to what we encountered in `vibe.d`?
A response for a timed out request could be waiting in the channel. Probably will never happen in practice.

File: source/geod24/concurrency.d [hunk 2]
```diff
@@ -463,6 +469,11 @@ public class FiberScheduler
         thisScheduler(this);
         scope (exit) thisScheduler(null);
 
+        /// Allow recursively calling dispatch
```
Comment (omerfirmak) [inline source/geod24/concurrency.d]: What is the use case for this?

**Review by omerfirmak — APPROVED** [review]

File: source/geod24/concurrency.d (hunk 2 above)
Comment (Geod24) [reply source/geod24/concurrency.d]: It was to allow an improvement for the scheduler in the main thread, but thinking about this more, I will take this commit out.

File: source/geod24/LocalRest.d (hunk 1 above)
Comment (Geod24) [reply source/geod24/LocalRest.d]: Hum that's actually a good point, I'll look at the code once more, and will take this commit out in the meantime.

**Pull request merged**